File: src/main/java/com/example/travelator/beansToValue/java/Application.java
package com.example.travelator.beansToValue.java;

public class Application {
	private final UserPreferences preferences;
	
	public Application(UserPreferences preferences) {
		this.preferences = preferences;
	}
	
	/**
	 * preferences 가 아니라 this를 넘겨야 함!!!!
	 */
	public void showWelcome() {
		new WelcomeView(preferences).show();
	}
	
	public void editPreferences() {
		new PreferencesView(preferences).show();
	}
}

File: src/main/java/com/example/travelator/beansToValue/Application.kt
package com.example.travelator.beansToValue

class Application(
    private var preferences: UserPreferences
) {
    fun showWelcome() {
        WelcomeView(preferences).show()
        WelcomeView(preferences).show()
    }

    fun editPreferences() {
        preferences = PreferencesView().showModal(preferences)
    }
}